src/profile-logic/committed-ranges.js

```diff
@@ -12,46 +12,126 @@ import type { StartEndRange } from 'firefox-profiler/types';
+        // the duration is 25ms. All values are integer.
+        // For microseconds: 12345678u25: the start is at 12'345'678µs, and the
+        // duration is 25µs.
+        // For nanoseconds: 12345678901n25: the start is at 12'345'678'901ns, and the
+        // duration is 25ns.
+        const m = s.match(/^(-?[0-9]+)([mun])([0-9]+)$/);
+        if (!m) {
+          // Flow doesn't like that we return null, but that's OK
+          // because we filter it out in the last operation.
+          // $FlowExpectError
+          return null;
+        }
+
+        // Let's convert values to milliseconds.
+        const start = Number(m[1]);
+        const durationUnit = m[2];
+        const duration = Number(m[3]);
+
+        let startInMs;
+        let endInMs;
+        switch (durationUnit) {
+          case 'm':
+            // values are in milliseconds.
+            startInMs = start;
+            endInMs = start + duration;
+            break;
+          case 'u':
+            // values are in microseconds.
+            startInMs = start / 1000;
+            endInMs = (start + duration) / 1000;
+            break;
+          case 'n':
+            // values are in nanoseconds.
+            startInMs = start / 1000000;
+            endInMs = (start + duration) / 1000000;
+            break;
+          default:
+            throw new Error(
+              `We couldn't recognize the unit ${durationUnit}, which can't happen because the regexp wouldn't match.`
+            );
+        }
 
-    const m1 = Number(m[1]);
-    let m2 = Number(m[2]);
+        if (startInMs === endInMs) {
+          // Duration could be 0 at first, or float operations rounding errors could
+          // also produce this: make sure that we have a non-empty range.
+          endInMs += 0.0001;
+        }
+
+        return { start: startInMs, end: endInMs };
+      })
+      // Filter out possible null values coming from bad inputs.
+      .filter(value => value)
+  );
+}
 
-    if (m2 === m1) {
-      // Ensure that the duration of the range is non-zero.
-      m2 = m2 + 0.0001;
+// This function returns a string representation of the pair { start, end } in
+// the form of the start + a duration: "<start><unit><duration>". We may lose
+// some precision, which is OK. We just need to take care that the resulting
+// range is close in length _and_ includes the requested range.
+// Note that start and end inputs are in milliseconds.
+export function stringifyStartEnd({ start, end }: StartEndRange): string {
+  // Let's work in integer nanoseconds for calculations, and with string
+  // manipulations, so that we avoid rounding errors due to floats.
+  const startInNs = Math.floor(start * 1000000);
+  const endInNs = Math.ceil(end * 1000000);
+
+  const durationInNs = endInNs - startInNs;
+  const strDurationInNs = String(durationInNs);
+
+  let result;
+  if (durationInNs > 9000000) {
+    // If the initial duration is more than 9ms, we'll output milliseconds
```

Comment:
this was the rationale: I want at least 2 integer digits.
That's why I used `> 9000000` => the idea was that they would be rounded to 10. See line 107.
I can write that in a comment.

Reply:
For example, if the duration is around 5ms, I felt that we should use µs. But if the duration is around 15ms we should use ms. I think that there's a range where this is debatable (let's say 5-20ms, and 5-20µs), but this isn't a big deal and can be changed later without changing the format otherwise, because the parsing function would work with different thresholds.

Reply:
Thanks, documenting the rationale is perfect.